pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamMessage.java

```diff
@@ -51,8 +51,8 @@ public StreamMessage(@Nullable byte[] key, T value, @Nullable StreamMessageMetad
     _length = length;
   }
 
-  public StreamMessage(T value, int length) {
-    this(null, value, null, length);
+  public StreamMessage(T value, int length, @Nullable StreamMessageMetadata metadata) {
+    this(null, value, metadata, length);
```

Comment:
let's keep original and add a nullable constructor?
```suggestion
  public StreamMessage(T value, int length) {
    this(value, length, null);
  }
  
  public StreamMessage(T value, int length, @Nullable StreamMessageMetadata metadata) {
    this(null, value, metadata, length);
```